cmd/jujud/reboot/reboot.go

```diff
@@ -164,3 +165,58 @@ func (r *Reboot) waitForContainersOrTimeout() error {
 		return errors.Trace(err)
 	}
 }
+
+func writeScript(args []string, after int) (string, error) {
+	tpl := `#!/bin/bash
+sleep %d
+%s`
+	script := fmt.Sprintf(tpl, after, strings.Join(args, " "))
+
+	f, err := tmpFile()
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	defer func() { _ = f.Close() }()
+
+	_, err = f.WriteString(script)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	name := f.Name()
+	err = os.Chmod(name, 0755)
```

Comment:
We're executing the file the actual interpreter so we don't need this script to be executable in fact it's probably better from a security stand point.